Backend/lib/scheduler/src/Lib/Scheduler/JobStorageType/Redis/Queries.hs

```diff
@@ -0,0 +1,105 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+
+module Lib.Scheduler.JobStorageType.Redis.Queries where
+
+import qualified Data.Aeson as DA
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.Text as T
+import Data.Text.Encoding as DT
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto as Esq
+import Kernel.Storage.Hedis
+import qualified Kernel.Storage.Hedis.Queries as Hedis
+import Kernel.Tools.Metrics.CoreMetrics
+import Kernel.Types.Common hiding (id)
+import Kernel.Types.Id
+import Kernel.Utils.Common (logDebug)
+import Kernel.Utils.Time (utcToMilliseconds)
+import Lib.Scheduler.Environment (SchedulerM)
+import Lib.Scheduler.JobStorageType.DB.Table (SchedulerJobT)
+import qualified Lib.Scheduler.ScheduleJob as ScheduleJob
+import Lib.Scheduler.Types
+
+createJob :: forall t (e :: t) m r. (JobFlow t e, JobMonad m, HedisFlow m r, HasField "schedulerSetName" r Text) => Int -> JobContent e -> m ()
+createJob maxShards jobData = do
+  void $ ScheduleJob.createJob @t @e createJobFunc maxShards $ JobEntry {jobData = jobData, maxErrors = 5}
+
+createJobIn :: forall t (e :: t) m r. (JobFlow t e, JobMonad m, HedisFlow m r, HasField "schedulerSetName" r Text) => NominalDiffTime -> Int -> JobContent e -> m ()
+createJobIn inTime maxShards jobData = do
+  void $ ScheduleJob.createJobIn @t @e createJobFunc inTime maxShards $ JobEntry {jobData = jobData, maxErrors = 5}
+
+createJobFunc :: (CoreMetrics m, HedisFlow m r, HasField "schedulerSetName" r Text) => AnyJob t -> m ()
+createJobFunc (AnyJob job) = do
+  key <- asks (.schedulerSetName)
+  Hedis.zAdd key [(utcToMilliseconds job.scheduledAt, AnyJob job)]
```

Comment:
lets use `utcToMilliseconds job.scheduledAt` as Id in the `Job` ? then we would not need to change the type signature of `getReadyTasks` function from `m [] to m ([], [])` as that is not backward compatible and also not a good way of doing it.